flutter/shell/platform/tizen/tizen_autofill.cc

```diff
@@ -0,0 +1,216 @@
+                      free(label);
+                    }
+
+                    return true;
+                  },
+                  group);
+              return true;
+            },
+            &count);
+        TizenAutofill::GetInstance().OnPopup();
+      },
+      nullptr);
+  if (ret != AUTOFILL_ERROR_NONE) {
+    FT_LOG(Error) << "Fail to set fill response received callback.";
+    return;
+  }
+
+  response_items_.clear();
+  initailzed_ = true;
+}
+
+void TizenAutofill::RequestAutofill(std::vector<std::string> hints,
+                                    std::string id) {
+  if (!initailzed_) {
+    Initailize();
+    return;
+  }
+
+  if (!connected_) {
+    return;
+  }
+
+  char* app_id = nullptr;
+  app_get_id(&app_id);
+
+  autofill_view_info_h view_info = nullptr;
+  autofill_view_info_create(&view_info);
+  autofill_view_info_set_app_id(view_info, app_id);
+  autofill_view_info_set_view_id(view_info, id.c_str());
+
+  if (app_id) {
+    free(app_id);
+  }
+
+  for (auto hint : hints) {
+    std::optional<autofill_hint_e> autofill_hint = ConvertAutofillHint(hint);
+    if (autofill_hint.has_value()) {
+      autofill_item_h item = nullptr;
+      autofill_item_create(&item);
+      autofill_item_set_autofill_hint(item, autofill_hint.value());
+      autofill_item_set_id(item, id.c_str());
+      autofill_item_set_sensitive_data(item, false);
+      autofill_view_info_add_item(view_info, item);
+      autofill_item_destroy(item);
+    }
+  }
+
+  int ret = autofill_fill_request(autofill_, view_info);
+  if (ret != AUTOFILL_ERROR_NONE) {
+    FT_LOG(Error) << "Fail to request autofill";
+  }
+  autofill_view_info_destroy(view_info);
+
+  response_items_.clear();
+}
+
+void TizenAutofill::RegisterItem(std::string view_id, AutofillItem item) {
+  if (!initailzed_) {
+    Initailize();
+    return;
+  }
+
+  if (!connected_) {
+    return;
+  }
+
+  autofill_save_item_h save_item = nullptr;
+  autofill_save_item_create(&save_item);
+  autofill_save_item_set_autofill_hint(save_item, item.hint_);
+  autofill_save_item_set_id(save_item, item.id_.c_str());
+  autofill_save_item_set_label(save_item, item.label_.c_str());
+  autofill_save_item_set_sensitive_data(save_item, item.sensitive_data_);
+  autofill_save_item_set_value(save_item, item.value_.c_str());
+
+  char* app_id = nullptr;
+  app_get_id(&app_id);
+
+  autofill_save_view_info_h save_view_info = nullptr;
+  autofill_save_view_info_create(&save_view_info);
+  autofill_save_view_info_set_app_id(save_view_info, app_id);
+  autofill_save_view_info_set_view_id(save_view_info, view_id.c_str());
+  autofill_save_view_info_add_item(save_view_info, save_item);
+
+  if (app_id) {
+    free(app_id);
+  }
+
+  int ret = autofill_commit(autofill_, save_view_info);
+  if (ret != AUTOFILL_ERROR_NONE) {
+    FT_LOG(Error) << "Fail to register autofill item.";
+  }
+
+  autofill_save_view_info_destroy(save_view_info);
+}
+
+std::optional<autofill_hint_e> TizenAutofill::ConvertAutofillHint(
```

Comment:
This method doesn't have to be a class member. Instead, you can create an anonymous namespace at the top of this file and move this function into it. (It's effectively the same as creating a C-style `static` function.)

Reply:
Created an anonoymous namespace at the top of `tizen_autofill.cc` and moved `ConvertAutofillHint` into it.